generate-profile.lic

```diff
@@ -0,0 +1,120 @@
+=begin
+This script takes partial YAML from profile-gen/ and resolves all %INCLUDE
+directives to produce new YAML files in profiles/ .
+
+Main yaml files must be named like
+Mychar-foo.yaml (where Mychar is your character name).
+The output files will have the same names as the these input files.
+
+Included yamls must either be named as:
+
+Mychar-foo-include.yaml (where Mychar is your character name), or
+ALL-foo-include.yaml
+
+In both cases these would be included into another YAML via:
+
+%INCLUDE foo
+
+See Risima-* in profile-gen for an example usage.
+=end
+
+class ProfileGenerator
+  def initialize(debug)
+    @debug = debug
+    load_profilegen
+    resolve_includes
+    write_profiles
+  end
+
+  def load_profilegen
+    @profiles = {}
+    paths = Dir['./scripts/profile-gen/*.yaml'].map { |item| File.path(item) }
+    paths.select! { |p| p.include?("#{checkname}-") || p.include?("ALL-") }
+    
+    paths.each do |path|
+      file = File.basename(path)
+      echo("Loading file #{file}") if @debug
+      unless match = /^(#{checkname}|ALL)-(.+?)(-include)?\.yaml$/.match(file)
+        raise ArgumentError, "Bad filename: #{file}"
+      end
+      name = match[2]
+      is_include = !!match[3]
+      if match[1] == 'ALL' && !is_include
+        raise ArgumentError, 'Files starting in ALL- must be includes.'
+      end
+      body = File.readlines("./scripts/profile-gen/#{file}")
```

Comment:
Does this script need to be trusted?

Reply:
Yes, it does.

Reply:
Could you add a note to the documentation?

Reply:
Done.